src/commands/check.rs

```diff
@@ -166,28 +223,35 @@ pub async fn check(args: CheckArgs) -> anyhow::Result<()> {
         };
 
         if !diagnostics.is_empty() {
-            emit_diagnostics(
-                diagnostics.iter().filter(|d| {
+            let filtered_diagnostics: Vec<&Diagnostic> = diagnostics
+                .iter()
+                .filter(|d| {
                     let severity = d.severity();
                     match severity {
                         Severity::Error => true,
                         Severity::Note if args.common.hide_notes => false,
                         _ if suppress => false,
                         _ => true,
                     }
-                }),
-                &uri,
-                &result.document().root().inner().text().to_string(),
-                args.common.report_mode,
-                args.common.no_color,
-            );
-
-            for diagnostic in diagnostics.iter() {
-                match diagnostic.severity() {
-                    Severity::Error => error_count += 1,
-                    Severity::Warning if !suppress => warning_count += 1,
-                    Severity::Note if !suppress => note_count += 1,
-                    _ => {}
+                })
+                .collect();
+
+            if !filtered_diagnostics.is_empty() {
+                emit_diagnostics(
+                    filtered_diagnostics.iter().copied(),
+                    &uri,
+                    &result.document().root().inner().text().to_string(),
+                    args.common.report_mode,
+                    args.common.no_color,
+                );
+
+                for diagnostic in diagnostics.iter() {
+                    match diagnostic.severity() {
+                        Severity::Error => error_count += 1,
+                        Severity::Warning if !suppress => warning_count += 1,
```

Comment:
I'm not sure, but I wonder if the "unknown except" warning should be counted here? Probably not, in which case we should add a comment here explicitly saying we don't add it to the count